cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramControllerServiceAdapter.java

```diff
@@ -80,13 +71,7 @@ public void running() {
       @Override
       public void failed(Service.State from, Throwable failure) {
         LOG.error("Program terminated with exception", failure);
-        if (propagateServiceError()) {
-          error(failure);
-        } else if (getState() != State.STOPPING) {
-          complete();
-        } else {
-          stop();
-        }
+        error(failure);
```

Comment:
Ohh ok. Got it.
No. It should not be `Service.State.STOPPING`. We are comparing the `ProgramController` states here and not the `Service` states.